testing/mock/test_system.cpp

```diff
@@ -433,6 +435,55 @@ int test_system::open(const std::string &path, int flags, mode_t mode) {
   return fd;
 }
 
+static bool _invalidate_read = false;
+static uint32_t _invalidate_read_after = 0;
+static const char * _invalidate_read_when_called_from = nullptr;
```

Comment:
I'm thinking we should make these variables (and the invalidate_malloc ones) members of test_system (in another PR of course). Thoughts?

Reply:
Yeah :)